File: TicketSellingSystem/Controllers/AdminUserController.cs
using System.Linq;
using System.Web;
using System.Web.Http;
using AutoMapper.QueryableExtensions;
using TicketSellingSystemInfrastructure.Constants;
using TicketSellingSystemInfrastructure.Exceptions;
using TicketSellingSystemInfrastructure.Models.ViewModels;
using TicketSellingSystemInfrastructure.Models.ViewModels.UserModels;
using TicketSellingSystemInfrastructure.Services;

namespace TicketSellingSystem.Controllers
{
    [Authorize(Roles = Roles.Admin)]
    public class AdminUserController : ApiController
    {
        private readonly IUserService _service;

        public AdminUserController(IUserService service)
        {
            _service = service;
        }

        [HttpGet]
        public PaginationGenericViewModel<AdminUserModel> GetUsers(int page)
        {
            var currUser = HttpContext.Current.User.Identity.Name;
            var users = _service
                .GetAllUsersExceptOne(currUser, page)
                .Project()
                .To<AdminUserModel>();
            return new PaginationGenericViewModel<AdminUserModel>
            {
                Count = _service.GetAllUsers().Count() - 1,
                CountPerPage = AppConstants.UsersPerPage,
                Items = users
            };
        }

        [HttpPost]
        public void DeleteUser([FromBody] LoginViewModel model)
        {
            try
            {
                _service.DeleteUser(model.Email);
            }
            catch (UserNotFoundException)
            {
            }
        }

        [HttpPost]
        public void BlockUser([FromBody] AdminUserModel model)
        {
            _service.BlockUser(model.Email, model.Blocked);
        }
    }
}

File: TicketSellingSystem/Scripts/Application/Controllers/adminUserController.js
(function(app) {
    'use strict';

    function AdminUserController($scope, users, adminUserService, userLoadingService, constants) {
        var vm = this;
        vm.users = users;
        vm.currentPage = constants.firstPage;
        vm.delete = function(email) {
            adminUserService.deleteUser(email);
            for (var i = 0, length = vm.users.Items.length; i < length; i++) {
                if (vm.users.Items[i].Email === email) {
                    vm.users.Items.splice(i, 1);
                    break;
                }
            }
        };

        vm.block = function(email) {
            for (var i = 0, length = vm.users.Items.length; i < length; i++) {
                if (vm.users.Items[i].Email === email) {
                    vm.users.Items[i].Blocked = !vm.users.Items[i].Blocked;
                    adminUserService.blockUser(email, vm.users.Items[i].Blocked);
                    break;
                }
            }
        };

        vm.isEmpty = function() {
            return vm.users.length === 0;
        };

        vm.pageChanged = function() {
            userLoadingService.getUsers(vm.currentPage)
                .then(function(response) {
                    vm.users = response;
                });
        };
    }

    app.controller('AdminUserController', ['$scope', 'users', 'adminUserService', 'userLoadingService', 'constants', AdminUserController]);
})(angular.module('controllers'));